### PR TITLE
Use path/directory/file name instead of path.

### PR DESCRIPTION
**Pull request by katmsft**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

Fixes https://github.com/Azure/azure-sdk-for-cpp/issues/1353

### CODE REVIEW

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_directory_client.hpp [hunk 1]
```diff
@@ -71,14 +71,14 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
...
-    DataLakeFileClient GetFileClient(const std::string& path) const;
+    DataLakeFileClient GetFileClient(const std::string& fileName) const;
 
     /**
      * @brief Create a DataLakeDirectoryClient from current DataLakeDirectoryClient
      * @param path Path of the directory under the current directory.
      * @return DataLakeDirectoryClient
      */
-    DataLakeDirectoryClient GetSubdirectoryClient(const std::string& path) const;
+    DataLakeDirectoryClient GetSubdirectoryClient(const std::string& subDirectoryName) const;
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_directory_client.hpp:81]: is `sub directory` one word or two words? We gotta be consistent about this

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_path_client.hpp [hunk 2]
```diff
@@ -34,7 +34,7 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
     static DataLakePathClient CreateFromConnectionString(
         const std::string& connectionString,
         const std::string& fileSystemName,
-        const std::string& path,
+        const std::string& pathName,
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_path_client.hpp:37]: pathname? what 's a path name?

Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_path_client.hpp:37]: Will change it into `dataLakePathName` to be more precise.

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_path_client.hpp:37]: No, it was `path` and you changed it to `pathName`. What is a path name? I know file name, directory name, they're comonly used words. I've hardly ever seen something called `path name`. Can you share some reference?

Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_path_client.hpp:37]: So what do you propose we call 'The name of a DataLake Path'?

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_path_client.hpp:37]: what's wrong with `path`?

Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_path_client.hpp:37]: I tend to believe that it is misleading but I don't feel strongly about it. If you think this is a good name I'm fine with it.

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_file_system_client.hpp [hunk 3]
```diff
@@ -70,24 +70,24 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
...
+    DataLakeFileClient GetFileClient(const std::string& fileName) const;
 
     /**
      * @brief Create a DataLakeDirectoryClient from current DataLakeFileSystemClient
-     * @param path Path of the directory within the file system.
+     * @param directoryName Name of the directory within the file system.
      * @return DataLakeDirectoryClient
      */
-    DataLakeDirectoryClient GetDirectoryClient(const std::string& path) const;
+    DataLakeDirectoryClient GetDirectoryClient(const std::string& directoryName) const;
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_file_system_client.hpp:90]: This is different how it's implemented in .Net. They only have `GetRootDirectoryClient`.

**Review by Jinming-Hu — CHANGES_REQUESTED** [review]
should we align with .Net?

File: sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_file_system_client.hpp (hunk 3 above)
Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_file_system_client.hpp:90]: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileSystemClient.cs#L469
I don't believe what you said is true.
The only thing not align is that .NET is not having GetPathClient. I don't think this inconsistency is being raised during the review so I'm sticking with this design now.

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_file_system_client.hpp:90]: okay, I missed that. So we need `GetRootDirectoryClient`.

Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_file_system_client.hpp:90]: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileSystemClient.cs#L496
.NET has this as internal. I don't think we should add that if we don't need that internally.

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_file_system_client.hpp:90]: Then it's okay

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_file_system_client.hpp:90]: But why don't they have `GetPathClient` anywhere?

Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_file_system_client.hpp:90]: Maybe they don't think it is important. But I tend to believe it is valuable for the customer to be able to manage the resource without knowing its type. If you take a look at the APIs supported by the path client, you would find that it is overkill to require the customer to identify the type of the resource before the API is called. e.g. GetProperties/SetMetadata.

Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_file_system_client.hpp:90]: Also, if `PathClient` supports initializing without the type of resource it points to, it is natural for its parent resource to support creating it.

**Review by Jinming-Hu — APPROVED** [review]

**Pull request merged**